### PR TITLE
`CountrySpecificPaymentFields`: refactor tests to `@testing-library/react`

### PR DESCRIPTION
**Pull request by flootr**
#### Proposed Changes

* `CountrySpecificPaymentFields`: refactor tests to `@testing-library/react`

#### Testing Instructions

* Verify unit tests still pass

### CODE REVIEW

File: client/my-sites/checkout/checkout/test/country-specific-payment-fields.js [hunk 1]
```diff
@@ -21,23 +24,38 @@ const defaultProps = {
...
-		const documentInput = wrapper.find( '[name="document"]' );
-		const event = { target: { name: 'document', value: 'spam' } };
-		documentInput.simulate( 'change', event );
-		expect( defaultProps.handleFieldChange ).toBeCalledWith( 'document', 'spam' );
+		render( <WrappedCountrySpecificPaymentFields { ...defaultProps } /> );
+
+		const box = screen.getByRole( 'textbox', { name: /taxpayer identification number/i } );
+		await user.type( box, 'spam' );
+
+		expect( defaultProps.handleFieldChange ).toHaveBeenNthCalledWith( 4, 'document', 'spam' );
```
Comment (tyxla) [inline client/my-sites/checkout/checkout/test/country-specific-payment-fields.js]: We could as well make this:

```suggestion
		expect( defaultProps.handleFieldChange ).toHaveBeenCalledWith( 'document', 'spam' );
```

to make it less specific if we don't care so much if it triggered 4 times or not (whether change triggering is debounced or not)

File: client/my-sites/checkout/checkout/test/country-specific-payment-fields.js [hunk 2]
```diff
@@ -21,23 +24,38 @@ const defaultProps = {
 };
 
 describe( '<CountrySpecificPaymentFields />', () => {
-	test( 'should render', () => {
-		const wrapper = shallow( <CountrySpecificPaymentFields { ...defaultProps } /> );
-		expect( wrapper ).toMatchSnapshot();
```
Comment (tyxla) [inline client/my-sites/checkout/checkout/test/country-specific-payment-fields.js:26]: Happy to see this removed 👍 I'm curious if there was a good reason to have it in the first place.

File: client/my-sites/checkout/checkout/test/country-specific-payment-fields.js [hunk 3]
```diff
@@ -21,23 +24,38 @@ const defaultProps = {
...
-	} );
+	test( 'should call this.props.handleFieldChange when updating field', async () => {
+		const user = userEvent.setup();
+
+		const WrappedCountrySpecificPaymentFields = ( props ) => {
```
Comment (tyxla) [inline client/my-sites/checkout/checkout/test/country-specific-payment-fields.js:30]: That's pretty elegant, nice 👍

File: client/my-sites/checkout/checkout/test/country-specific-payment-fields.js [hunk 4]
```diff
@@ -21,23 +24,38 @@ const defaultProps = {
...
 	} );
 
 	test( 'should disable fields', () => {
-		const wrapper = shallow( <CountrySpecificPaymentFields { ...defaultProps } /> );
-		expect( wrapper.find( 'Input' ).first().props().disabled ).toEqual( false );
-		wrapper.setProps( { disableFields: true } );
-		expect( wrapper.find( 'Input' ).first().props().disabled ).toEqual( true );
+		const { rerender } = render( <CountrySpecificPaymentFields { ...defaultProps } /> );
+
+		const [ input ] = screen.getAllByRole( 'textbox' );
```
Comment (tyxla) [inline client/my-sites/checkout/checkout/test/country-specific-payment-fields.js]: I know the previous test was checking just the first field for being disabled, but I wonder if we should check them all 🤔

**Review by tyxla — APPROVED** [review]
LGTM from my perspective 👍 🚢

File: client/my-sites/checkout/checkout/test/country-specific-payment-fields.js (hunk 1 above)
Comment (flootr) [reply client/my-sites/checkout/checkout/test/country-specific-payment-fields.js]: Agreed 👍🏻

**Pull request merged**